autorest/azure/token.go

```diff
@@ -215,23 +256,25 @@ func (spt *ServicePrincipalToken) Refresh() error {
 
 	v := url.Values{}
 	v.Set("client_id", spt.clientID)
-	v.Set("grant_type", "client_credentials")
 	v.Set("resource", spt.resource)
 
-	err := spt.secret.SetAuthenticationValues(spt, &v)
-	if err != nil {
-		return err
+	if spt.RefreshToken != "" {
+		v.Set("grant_type", OAuthGrantTypeRefreshToken)
```

Comment:
On one hand, I could have a flag, or do a type assertion. On the other hand, I'd prefer to use refreshtoken anytime it's available and it will always be valid to do so. I lean towards leaving as-is.